### DISCUSSION TITLE
Show full name of a user from people picker field

### DISCUSSION THREAD

**Discussion by AnouckF** [microsoft-search/pnp-modern-search#3243] [Q&A]
Can I use a regex expression in the managed columns Edit Handlebars expression?

When I go to debug mode, I see the value is the following for the field I'm willing to use  

`"user@email.com | User fullname | 693A30232E667C6D656D626572736869707C615F64617669642E6D616E6E616572744064696769706F6C69732E6F6E6D6963726F736F66742E636F6D i:0#.f|membership|user@email.com"`

I only would like to show the value between the first and second |

**Comment by kasperbolarsen** [comment]
I can't see which Managed Property you are using. If the Person or Group site column is mapped to a RefinableString you will automatically get the Users name displayed, at least in the Details List and List layouts.

**Reply by AnouckF** [reply]
I don't use a managed property but a site column. It's a Person field

**Reply by kasperbolarsen** [reply]
I guarantee that you are using a Managed Property as those are the only properties that can be using in search :-) Your site colluction field gets indexed and this creates a Crawled Property (and sometimes an automatically Managed Property). You can map a RefinableString or RefinableInt (depending on the Date type) to the crawled property, if you need to be able to sort or refine on that property.
Are you currently using the OWS_[Name of your field] or the OWS_Q_USER_[Name of your field] ?

**Reply by kasperbolarsen** [reply]
Yes, you can use Regex in the template as we have the Regex Handlebar helper , see https://microsoft-search.github.io/pnp-modern-search/v3/search-parts/templating/#handlebars-helpers
Or you can use something like this:
<span class="template--listItem--author">
                                            {{#with (split (slot item @root.slots.Author) '|')}}
                                                {{[1]}}
                                            {{/with}}
                                        </span>
